### PR TITLE
[akeneo-design-system]: moving to rollup module bundler

### PR DESCRIPTION
**Pull request by nicomaligne**
**Description (for Contributor and Core Developer)**
Removing webpack module bundler in favor of rollup.
Different articles explaining the why of this change 
https://medium.com/webpack/webpack-and-rollup-the-same-but-different-a41ad427058c
https://stackoverflow.com/questions/43219030/what-is-flat-bundling-and-why-is-rollup-better-at-this-than-webpack
https://medium.com/jsdownunder/rollup-vs-webpack-javascript-bundling-in-2018-b35758a2268
https://nolanlawson.com/2016/08/15/the-cost-of-small-modules/
These articles are old but explained the build differences between webpack and rollup.
Rollup delivers a flat file wit no IFEE function and caching like webpack. 
Webpack have a plugin to mimic the rollup build nowadays.
But still rollup is the recommended choice to build a library (used to build react / vue and others). 

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |-
| Added legacy Behats               |-
| Added acceptance tests            |-
| Added integration tests           |-
| Changelog updated                 |-
| Review and 2 GTM                  |-
| Micro Demo to the PO (Story only) |-
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: akeneo-design-system/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import commonjs from 'rollup-plugin-commonjs';
+import babel from 'rollup-plugin-babel';
+import analyze from 'rollup-plugin-analyzer';
+import resolve from 'rollup-plugin-node-resolve';
+import {terser} from 'rollup-plugin-terser';
+import replace from '@rollup/plugin-replace';
+
+import pkg from './package.json';
+
+const extensions = ['.js', '.ts', '.tsx'];
```
Comment (pchasle) [inline akeneo-design-system/rollup.config.js]:
```suggestion
const extensions = ['.ts', '.tsx'];
```

File: akeneo-design-system/package.json [hunk 2]
```diff
@@ -2,32 +2,34 @@
...
         "check-types": "tsc",
         "check-types:watch": "yarn check-types -- --watch",
-        "build:dev": "yarn prebuild && webpack --mode development",
-        "build:prod": "yarn prebuild && webpack --mode production",
         "test": "jest",
         "test:watch": "jest --watch",
         "storybook": "start-storybook -p 6006",
         "build-storybook": "build-storybook",
-        "build:analyzer": "yarn build:prod --env.analyzer"
+        "build:prod": "yarn prebuild && export NODE_ENV=production && rollup -c",
```
Comment (tseho) [inline akeneo-design-system/package.json]:
```suggestion
        "build:prod": "yarn prebuild && NODE_ENV=production rollup -c",
```

**Comment by juliensnz** [comment]
What are the benefits of rollup for us? Is it faster? does it reduce the size of the package? I am not against change, but it would be nice to really explain why it would be interesting for us :)

**Comment by tseho** [comment]
as discussed IRL, I would love to have a test on what are the impact of creating a bundle of the lib on:
- source map
- tree shaking

**Comment by tseho** [comment]
regarding the fact that rollup includes react in the bundle, I don't think that's a good idea, the included react can have a mismatched version with `react-dom` bundled by the application.

**Pull request closed without merging**